cartridges/int_algolia/cartridge/scripts/algolia/steps/sendChunkOrientedDeltaProductUpdates.js

```diff
@@ -0,0 +1,336 @@
+
+// job step parameters
+var paramConsumer, paramDeltaExportJobName, paramFieldListOverride;
+
+// Algolia requires
+var algoliaData, AlgoliaLocalizedProduct, algoliaProductConfig, jobHelper, fileHelper, algoliaIndexingAPI, sendHelper, productFilter;
+
+// logging-related variables and constants
+var logData;
+const updateLogType = 'LastProductDeltaSyncLog';
+const resourceType = 'productdelta';
+
+var l0_deltaExportDir, l1_processingDir, l1_completedDir;
+var changedProducts = [], changedProductsIterator;
+var deltaExportZips, siteLocales, nonLocalizedAttributes = [], fieldsToSend;
+var baseIndexingOperation; // 'addObject' or 'partialUpdateObject', depending on the step parameter 'fullRecordUpdate'
+const deleteIndexingOperation = 'deleteObject';
+
+/*
+ * Rough algorithm of chunk-oriented script module execution:
+ *
+ *  counter = 0
+ *  beforeStep() - executed before starting the job step, initialization
+ *    repeat until counter reaches getTotalCount() {
+ *         create new thread {
+ *             beforeChunk() - executed before each chunk
+ *             read() - repeat until chunkSize, each run reads one product
+ *             process() - repeat until chunkSize, each run processes one product
+ *             write() - executed one time, sends all products from the chunk at once
+ *             counter += chunkSize
+ *             afterChunk() - executed after each chunk
+ *         }
+ *     }
+ *     afterStep() - executed after the job step finishes
+ */
+
+/**
+ * before-step-function (steptypes.json)
+ * Any returns from this function result in skipping to the afterStep() function (omitting read-process-writealtogether)
+ * with the "success" parameter passed to it set to false.
+ * @param {dw.util.HashMap} parameters job step parameters
+ * @param {dw.job.JobStepExecution} stepExecution contains information about the job step
+*/
+exports.beforeStep = function(parameters, stepExecution) {
+    algoliaData = require('*/cartridge/scripts/algolia/lib/algoliaData');
+    AlgoliaLocalizedProduct = require('*/cartridge/scripts/algolia/model/algoliaLocalizedProduct');
+    algoliaProductConfig = require('*/cartridge/scripts/algolia/lib/algoliaProductConfig');
+    jobHelper = require('*/cartridge/scripts/algolia/helper/jobHelper');
+    fileHelper = require('*/cartridge/scripts/algolia/helper/fileHelper');
+    algoliaIndexingAPI = require('*/cartridge/scripts/algoliaIndexingAPI');
+    logger = require('dw/system/Logger').getLogger('algolia', 'Algolia');
+    productFilter = require('*/cartridge/scripts/algolia/filters/productFilter');
+    sendHelper = require('*/cartridge/scripts/algolia/helper/sendHelper');
+
+    // checking mandatory parameters
+    if (empty(parameters.consumer) || empty(parameters.deltaExportJobName)) {
+        let errorMessage = 'Mandatory job step parameters missing!';
+        jobHelper.logError(errorMessage);
+        return;
+    }
+
+    // parameters
+    paramConsumer = parameters.consumer.trim();
```

Comment:
If it's only used for the export dir path, what about hardcoding it and remove the parameter?

Reply:
It's not, the same parameter and the same value need to be configured for the B2C delta export job step.

Reply:
Right I meant we can have a constant in the code that both jobs uses. We can discuss that outside of this PR.

Reply:
We can't do that as we wouldn't be able to use that constant for the B2C job. It's a common parameter of both the B2C delta job step and the chunk-based second step and the value needs to be the same. The problem is that since we don't control the code of the B2C job, the only way to interface with it is its the parameters, we couldn't use a constant from our code.

Reply:
Oh ok, I didn't get that it was a built-in job, nevermind!